labelbox/data/mixins.py

```diff
@@ -1,32 +1,70 @@
-from typing import Optional
+from typing import Optional, List
 
 from pydantic import BaseModel, validator
 
-from labelbox.exceptions import ConfidenceNotSupportedException
+from labelbox.exceptions import ConfidenceNotSupportedException, CustomMetricsNotSupportedException
 
 
 class ConfidenceMixin(BaseModel):
     confidence: Optional[float] = None
 
-    @validator('confidence')
+    @validator("confidence")
     def confidence_valid_float(cls, value):
         if value is None:
             return value
         if not isinstance(value, (int, float)) or not 0 <= value <= 1:
-            raise ValueError('must be a number within [0,1] range')
+            raise ValueError("must be a number within [0,1] range")
         return value
 
     def dict(self, *args, **kwargs):
         res = super().dict(*args, **kwargs)
-        if 'confidence' in res and res['confidence'] is None:
-            res.pop('confidence')
+        if "confidence" in res and res["confidence"] is None:
+            res.pop("confidence")
         return res
 
 
 class ConfidenceNotSupportedMixin:
-
     def __new__(cls, *args, **kwargs):
-        if 'confidence' in kwargs:
+        if "confidence" in kwargs:
             raise ConfidenceNotSupportedException(
-                'Confidence is not supported for this annotaiton type yet')
+                "Confidence is not supported for this annotation type yet"
+            )
+        return super().__new__(cls)
+
+
+class CustomMetric(BaseModel):
+    name: str
+    value: float
+
+    @validator("name")
```

Comment:
yep this is a good catch... pydantic will coerce to str any other possible type... I know in pydantic2 there is a config to make it more strict.  I did not see anything like this in pydantic 1. Good catch